release notes/v0.40.0.md

```diff
@@ -0,0 +1,280 @@
+
+Additionally, as previously mentioned, there were a bunch of changes to goja, including adding class support which also needed to be integrated.
+
+A future breaking change () is that it will no longer be possible to use CommonJS style exports along import/export syntax in the *same* file.
+
+```javascript
+import http from "k6/http"
+
+exports.default = function() {} // this will start to error out
+```
+
+It will still be possible to import a CommonJS module and `require` an ES module or use `require` in an ES module. Having a dependency cycle mixing and matching CommonJS and ESM is also unlikely to work properly, but *might* do so in particular cases.
+
+This is really expected to have never been done by anyone as there isn't really much of a reason for this, but it is currently supported due to Babel transpiling everything to CommonJS behind the scenes.
+
+## Refactoring metrics
+
+The refactoring of metrics is underway, with a [PR](https://github.com/grafana/k6/pull/2594) for more performant internal representation. Unfortunately, this took longer to get to a stable state, and given the size and the type of change, the k6 team decided to *hold* merging it until very near the end of the cycle. It also doesn't have any noticeable change for most users. Instead, it will be merged early in the v0.41.0 release cycle, and then more changes to use it will be made through the release cycle.
+
+Some of those changes include supporting non-indexable tags, which also have a read [PR](https://github.com/grafana/k6/pull/2654). This change is also essential for historical reasons connected with how the `name`  tag works. As such, it also needed to be merged to release the internal metric refactor.
+
+Future breaking change: As part of the many changes and experiments, we found out that we can keep `url` as an indexable tag. Previously the plan was that it *along* `vu` and `iter` will become non-indexable. However, the `url` tag is *heavily* used and enabled by default. Because of that (and other internal reasons), the new change will be that `url` will stay indexable, but if used with `name` will be overwritten to have the same value as `name`. Arguably this is what most users would want in the case when they are using `name`. We plan to add a *non* indexable `raw_url` tag for those that do not. As such, we no longer will be [printing a warning when `url` is used](https://github.com/grafana/k6/pull/2655) in thresholds.
+
+Even as such, we did make a bunch of changes to the internals of k6 that will pave the way forward. [#2629](https://github.com/grafana/k6/pull/2629), [#2631](https://github.com/grafana/k6/pull/2631).
+
+[Prometheus remote-write output extension](https://github.com/grafana/xk6-output-prometheus-remote/) was on hold as it heavily depends on metrics' refactoring.
```

Comment:
```suggestion
[Prometheus remote-write output extension](https://github.com/grafana/xk6-output-prometheus-remote) we are still working on the optimal solution using the new developed time series data model that will add an important performance improvement for the Prometheus RW output and it will simplify the process to include the output in the k6 binary, but in the meanwhile, we have introduced a simplified implementation that adds the high requested ability to collect and flush metrics by time series (the combination between the metric and the possible tag sets). It has the benefit to facilitate the metric aggregation reducing the flushed samples, and as a consequence, the pressure on the Prometheus Remote write endpoint.
Also, a bug has been fixed where multiple samples were flushed with the same timestamp generating several warnings on the more advanced implementations of the Remote write protocol like [Mimir](https://grafana.com/oss/mimir).
```


Here, my suggestion for the Prometheus RW output. If we are going to add it I will tag the new version.

Reply:
I am not certain given that this is based on unmerged branch of k6 - how nice it is for us to release version of an extension pointing to a not `master` branch commit. If we rebase the branch when merging, the commit being pointed to will suddenlyt not exist and consequently ... I guess it will only work while the go cache is still containing it. 

So in general I would prefer to not do this. It will be w/e for some experimental extension where we don't care about it working, but this one seems important enough for us to not want to have broken versions.

WDYT @codebien @na--

Reply:
As mentioned in private, this is mostly related to an ad-hoc change based on the current model in the master and not related to the new data model we are developing.

For this reason, I wrote:

>the optimal solution using the new developed time series data model that will add an important performance improvement for the Prometheus RW output

In this way, we can collect feedback on the real logic for the Prometheus output where metrics are aggregated by time series and not only by the partial use case with only the main metric.

WDYT?

Reply:
we did went with a shorter version https://github.com/grafana/k6/pull/2669#discussion_r964705988